# Project Overview: Business Survey Data Analysis

This project focuses on the **cleaning, management, and analysis of business survey data collected in Migori during the 2025/2026 field study**. The dataset was gathered through a structured questionnaire administered by enumerators, each tasked with collecting responses from three businesses within the study area. The survey aimed to capture information on the **characteristics of local enterprises, ownership structures, employment size, operational history, and the challenges faced by businesses**.  

The workflow of this project begins with **data preparation using SQL**, where the raw survey dataset is cleaned and structured for analysis. This process involves standardizing column names, correcting inconsistent responses, handling missing values, verifying enumerator submissions, and organizing categorical variables into a format suitable for analysis. Care is taken to preserve all original observations since each record represents a unique business surveyed in the field.  

Once cleaned, the dataset is transformed through **aggregation queries** to produce summary statistics such as the distribution of businesses by sector, ownership category, gender of business owners, years of operation, and the major constraints affecting enterprises. These aggregations help convert raw questionnaire responses into meaningful analytical indicators.  

The cleaned and structured data is then connected to **Microsoft Power BI**, where interactive dashboards are developed to visualize the results. The dashboards provide insights into the **local economic structure, business diversity, employment patterns, and operational challenges faced by small enterprises** in the study area. Visualizations such as bar charts, pie charts, and key performance indicators enable clear communication of patterns observed in the data.  

Overall, this project demonstrates how **field-collected survey data can be transformed into actionable insights through data cleaning, SQL-based aggregation, and data visualization tools**. The final dashboard serves as a decision-support resource that can assist planners, researchers, and local stakeholders in understanding the **business environment and identifying areas for economic support and policy intervention**.


### NOTE: Oruba has a very small sample size in this output only 3 records which might skew the comparison as they survey concentrated on the A1 road

### installing the required tools

In [1]:
pip install ipython-sql pymysql

Note: you may need to restart the kernel to use updated packages.


In [4]:
%load_ext sql
%config SqlMagic.style = '_DEPRECATED_DEFAULT'
%config SqlMagic.autopandas = True

In [5]:
%sql mysql+pymysql://root:********@localhost:3306/business_data

### Previewing Our Data

In [6]:
%%sql
SELECT*
FROM business_data
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/business_data
5 rows affected.


,Serial #,Researcher Name,Adm. No,Date,Place,Gender,Age Group,Education,Role (Owner/Emp),Ownership_Category,...,AFN_Activities,Industrial_Factories,High_cost,Taxes,Acces_finance,Access_premises,Infrastructure_gaps,Sanitation,Competition,Other Issues
0,1,Bravin,B15/7604/2023,2/11/2026,Lower Marindi,Male,18 - 25 Yrs,Secondary,Owner,Partnership,...,N/A,N/A,N/A,N/A,1,N/A,2,N/A,3,N/A
1,2,Bravin,B15/7604/2023,2/11/2026,Lower Marindi,Female,36 - 45 Yrs,No Formal Education,Owner,Sole Proprietor,...,N/A,N/A,2,1,N/A,N/A,N/A,3,N/A,N/A
2,3,Bravin,B15/7604/2023,2/11/2026,Lower Marindi,Female,36 - 45 Yrs,No Formal Education,Owner,Sole Proprietor,...,N/A,N/A,2,N/A,1,N/A,N/A,3,N/A,N/A
3,1,Cornel,B15/6451/2021,11/2/2021,Lower Marindi,Female,36 - 45 Yrs,Secondary,Employee,Partnership,...,N/A,N/A,N/A,3,1,N/A,N/A,2,N/A,N/A
4,2,Cornel,B15/6451/2021,11/2/2021,Lower Marindi,Female,26 - 35 Yrs,College,Employee,Partnership,...,N/A,N/A,N/A,2,N/A,N/A,N/A,1,3,N/A


### Describing our Tables
Understand our Data Formats

In [7]:
%sql DESCRIBE business_data

 * mysql+pymysql://root:***@localhost:3306/business_data
32 rows affected.


,Field,Type,Null,Key,Default,Extra
0,Serial #,int,YES,,None,
1,Researcher Name,text,YES,,None,
2,Adm. No,text,YES,,None,
3,Date,text,YES,,None,
4,Place,text,YES,,None,
5,Gender,text,YES,,None,
6,Age Group,text,YES,,None,
7,Education,text,YES,,None,
8,Role (Owner/Emp),text,YES,,None,
9,Ownership_Category,text,YES,,None,


## Data Cleaning
### Dropping Irrelevant Columns
Serial,Reasearcher Name, Date as the survey was done on same day, Admission No

In [9]:
%%sql
ALTER TABLE business_data
DROP COLUMN `Researcher Name`,
DROP COLUMN `Adm. No`,
DROP COLUMN `Date`;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.


""


In [11]:
%%sql
SELECT*
FROM business_data
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/business_data
5 rows affected.


,Serial #,Place,Gender,Age Group,Education,Role (Owner/Emp),Ownership_Category,Employee_Size,Years_Operated,retail_services,...,AFN_Activities,Industrial_Factories,High_cost,Taxes,Acces_finance,Access_premises,Infrastructure_gaps,Sanitation,Competition,Other Issues
0,1,Lower Marindi,Male,18 - 25 Yrs,Secondary,Owner,Partnership,1-5 employees,6-10 yrs,N/A,...,N/A,N/A,N/A,N/A,1,N/A,2,N/A,3,N/A
1,2,Lower Marindi,Female,36 - 45 Yrs,No Formal Education,Owner,Sole Proprietor,1-5 employees,1-5 yrs,Hardware,...,N/A,N/A,2,1,N/A,N/A,N/A,3,N/A,N/A
2,3,Lower Marindi,Female,36 - 45 Yrs,No Formal Education,Owner,Sole Proprietor,1-5 employees,1-5 yrs,Hardware,...,N/A,N/A,2,N/A,1,N/A,N/A,3,N/A,N/A
3,1,Lower Marindi,Female,36 - 45 Yrs,Secondary,Employee,Partnership,1-5 employees,More than 10 yrs,Boutique/Exhibition,...,N/A,N/A,N/A,3,1,N/A,N/A,2,N/A,N/A
4,2,Lower Marindi,Female,26 - 35 Yrs,College,Employee,Partnership,1-5 employees,1-5 yrs,None,...,N/A,N/A,N/A,2,N/A,N/A,N/A,1,3,N/A


In [13]:
%%sql
ALTER TABLE business_data
DROP COLUMN `Serial #`;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.


""


In [14]:
%%sql
SELECT*
FROM business_data
LIMIT 5

 * mysql+pymysql://root:***@localhost:3306/business_data
5 rows affected.


,Place,Gender,Age Group,Education,Role (Owner/Emp),Ownership_Category,Employee_Size,Years_Operated,retail_services,Informal_Sector,...,AFN_Activities,Industrial_Factories,High_cost,Taxes,Acces_finance,Access_premises,Infrastructure_gaps,Sanitation,Competition,Other Issues
0,Lower Marindi,Male,18 - 25 Yrs,Secondary,Owner,Partnership,1-5 employees,6-10 yrs,N/A,Welding,...,N/A,N/A,N/A,N/A,1,N/A,2,N/A,3,N/A
1,Lower Marindi,Female,36 - 45 Yrs,No Formal Education,Owner,Sole Proprietor,1-5 employees,1-5 yrs,Hardware,N/A,...,N/A,N/A,2,1,N/A,N/A,N/A,3,N/A,N/A
2,Lower Marindi,Female,36 - 45 Yrs,No Formal Education,Owner,Sole Proprietor,1-5 employees,1-5 yrs,Hardware,N/A,...,N/A,N/A,2,N/A,1,N/A,N/A,3,N/A,N/A
3,Lower Marindi,Female,36 - 45 Yrs,Secondary,Employee,Partnership,1-5 employees,More than 10 yrs,Boutique/Exhibition,None,...,N/A,N/A,N/A,3,1,N/A,N/A,2,N/A,N/A
4,Lower Marindi,Female,26 - 35 Yrs,College,Employee,Partnership,1-5 employees,1-5 yrs,None,None,...,N/A,N/A,N/A,2,N/A,N/A,N/A,1,3,N/A


## Standardizing Values
### Gender Standardization

In [17]:
%%sql
UPDATE business_data
SET Gender = 'Male'
WHERE Gender IN ('male','m','M');

UPDATE business_data
SET Gender = 'Female'
WHERE Gender IN ('female','f','F');

 * mysql+pymysql://root:***@localhost:3306/business_data
26 rows affected.
40 rows affected.


""


In [18]:
%%sql
SELECT*
FROM business_data
LIMIT 5

 * mysql+pymysql://root:***@localhost:3306/business_data
5 rows affected.


,Place,Gender,Age Group,Education,Role (Owner/Emp),Ownership_Category,Employee_Size,Years_Operated,retail_services,Informal_Sector,...,AFN_Activities,Industrial_Factories,High_cost,Taxes,Acces_finance,Access_premises,Infrastructure_gaps,Sanitation,Competition,Other Issues
0,Lower Marindi,Male,18 - 25 Yrs,Secondary,Owner,Partnership,1-5 employees,6-10 yrs,N/A,Welding,...,N/A,N/A,N/A,N/A,1,N/A,2,N/A,3,N/A
1,Lower Marindi,Female,36 - 45 Yrs,No Formal Education,Owner,Sole Proprietor,1-5 employees,1-5 yrs,Hardware,N/A,...,N/A,N/A,2,1,N/A,N/A,N/A,3,N/A,N/A
2,Lower Marindi,Female,36 - 45 Yrs,No Formal Education,Owner,Sole Proprietor,1-5 employees,1-5 yrs,Hardware,N/A,...,N/A,N/A,2,N/A,1,N/A,N/A,3,N/A,N/A
3,Lower Marindi,Female,36 - 45 Yrs,Secondary,Employee,Partnership,1-5 employees,More than 10 yrs,Boutique/Exhibition,None,...,N/A,N/A,N/A,3,1,N/A,N/A,2,N/A,N/A
4,Lower Marindi,Female,26 - 35 Yrs,College,Employee,Partnership,1-5 employees,1-5 yrs,None,None,...,N/A,N/A,N/A,2,N/A,N/A,N/A,1,3,N/A


### Catering For Extra Spaces 
By Trimming

In [35]:
%%sql
Update business_data
SET 
    `Place`=TRIM(`Place`),
    `Gender`=TRIM(`Gender`),
    `Age Group`=TRIM(`Age Group`),
    `Education`=TRIM(`Education`),
    `Role (Owner/Emp)`=TRIM(`Role (Owner/Emp)`),
    `Ownership_Category`=TRIM(`Ownership_Category`),
    `Employee_Size`=TRIM(`Employee_Size`),
    `Years_Operated`=TRIM(`Years_Operated`),
    `retail_services`=TRIM(`retail_services`),
    `Informal_Sector`=TRIM(`Informal_Sector`),
    `Accomodation_Catering`=TRIM(`Accomodation_Catering`),		
    `PTF_Services`=TRIM(`PTF_Services`),
    `Education_Institution`=TRIM(`Education_Institution`),
    `Health`=TRIM(`Health`),
    `Entertainment`=TRIM(`Entertainment`),
    `Transport`=TRIM(`Transport`),	
    `Storage`=TRIM(`Storage`),
    `Communications`=TRIM(`Communications`),
    `AFN_Activities`=TRIM(`AFN_Activities`),
    `Industrial_Factories`=TRIM(`Industrial_Factories`),
    `High_cost`=TRIM(`High_cost`),
    `Taxes`=TRIM(`Taxes`),
    `Acces_finance`=TRIM(`Acces_finance`),
    `Access_premises`=TRIM(`Access_premises`),
    `Infrastructure_gaps`=TRIM(`Infrastructure_gaps`),
    `Sanitation`=TRIM(`Sanitation`),
    `Competition`=TRIM(`Competition`),
    `Other Issues`=TRIM(`Other Issues`);


 * mysql+pymysql://root:***@localhost:3306/business_data
66 rows affected.


""


### Replacing None with N/A for Uniformity & High_cost Yes with Value

In [50]:
%%sql
UPDATE business_data
SET Informal_Sector='N/A'
WHERE Informal_Sector='None';
# High_cost col
UPDATE business_data
SET High_cost='1'
WHERE High_cost='yes';
# Retail Services
UPDATE business_data
SET retail_services='N/A'
WHERE retail_services='None';




 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
0 rows affected.
1 rows affected.


""


In [51]:
%%sql
SELECT*
FROM business_data
LIMIT 5

 * mysql+pymysql://root:***@localhost:3306/business_data
5 rows affected.


,Place,Gender,Age Group,Education,Role (Owner/Emp),Ownership_Category,Employee_Size,Years_Operated,retail_services,Informal_Sector,...,Industrial_Factories,High_cost,Taxes,Acces_finance,Access_premises,Infrastructure_gaps,Sanitation,Competition,Other Issues,id
0,Lower Marindi,Male,18 - 25 Yrs,Secondary,Owner,Partnership,1-5 employees,6-10 yrs,N/A,Welding,...,N/A,N/A,N/A,1,N/A,2,N/A,3,N/A,1
1,Lower Marindi,Female,36 - 45 Yrs,No Formal Education,Owner,Sole Proprietor,1-5 employees,1-5 yrs,Hardware,N/A,...,N/A,2,1,N/A,N/A,N/A,3,N/A,N/A,2
2,Lower Marindi,Female,36 - 45 Yrs,No Formal Education,Owner,Sole Proprietor,1-5 employees,1-5 yrs,Hardware,N/A,...,N/A,2,N/A,1,N/A,N/A,3,N/A,N/A,3
3,Lower Marindi,Female,36 - 45 Yrs,Secondary,Employee,Partnership,1-5 employees,More than 10 yrs,Boutique/Exhibition,N/A,...,N/A,N/A,3,1,N/A,N/A,2,N/A,N/A,4
4,Lower Marindi,Female,26 - 35 Yrs,College,Employee,Partnership,1-5 employees,1-5 yrs,N/A,N/A,...,N/A,N/A,2,N/A,N/A,N/A,1,3,N/A,5


## Normalization
Data normalization for business categories

Adding an Id column to serve as my primary key

In [57]:
%%sql
ALTER TABLE business_data
ADD id INT AUTO_INCREMENT PRIMARY KEY;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.


""


In [58]:
%%sql
CREATE TABLE IF NOT EXISTS business_sector_activity (
    id INT AUTO_INCREMENT PRIMARY KEY,
    business_id INT,
    sector VARCHAR(150),
    activity VARCHAR(150)
);

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.


""


In [64]:
%%sql
# Retail Sector Data
INSERT INTO business_sector_activity (business_id, sector, activity)
SELECT id, 'General Traders, Shop & Retail Services', retail_services
FROM business_data
WHERE retail_services IS NOT NULL
AND retail_services <> 'N/A'
AND retail_services='None';

# Informal Sector Data
INSERT INTO business_sector_activity (business_id, sector, activity)
SELECT id, 'Informal Sector', Informal_Sector
FROM business_data
WHERE Informal_Sector IS NOT NULL
AND Informal_Sector <> 'N/A'
AND Informal_Sector <> 'None';

# Accommodation & Catering
INSERT INTO business_sector_activity (business_id, sector, activity)
SELECT id, 'Accommodation & Catering', Accomodation_Catering
FROM business_data
WHERE Accomodation_Catering IS NOT NULL
AND Accomodation_Catering<>'N/A'
AND Accomodation_Catering<>'None';

# Professional / Technical / Financial Services
INSERT INTO business_sector_activity (business_id, sector, activity)
SELECT id, 'Professional, Technical & Financial Services', PTF_Services
FROM business_data
WHERE PTF_Services IS NOT NULL
AND PTF_Services<>'N/A'
AND PTF_Services<>'None';

# Insert Education Institutions
INSERT INTO business_sector_activity (business_id, sector, activity)
SELECT id, 'Education Institution', Education_Institution
FROM business_data
WHERE Education_Institution IS NOT NULL
AND Education_Institution<>'N/A'
AND Education_Institution<>'None';

# Health
INSERT INTO business_sector_activity (business_id, sector, activity)
SELECT id, 'Health', Health
FROM business_data
WHERE Health IS NOT NULL
AND Health<>'N/A'
AND Health<>'None';

# Entertainment
INSERT INTO business_sector_activity (business_id, sector, activity)
SELECT id, 'Entertainment', Entertainment
FROM business_data
WHERE Entertainment IS NOT NULL
AND Entertainment<>'N/A'
AND Entertainment<>'None';

# Transport
INSERT INTO business_sector_activity (business_id, sector, activity)
SELECT id, 'Transport', Transport
FROM business_data
WHERE Transport IS NOT NULL
AND Transport<>'N/A'
AND Transport<>'None';

# Storage
INSERT INTO business_sector_activity (business_id, sector, activity)
SELECT id, 'Storage', Storage
FROM business_data
WHERE Storage IS NOT NULL
AND Storage<>'N/A'
AND Storage<>'None';

# Communications
INSERT INTO business_sector_activity (business_id, sector, activity)
SELECT id, 'Communications', Communications
FROM business_data
WHERE Communications IS NOT NULL
AND Communications<>'N/A'
AND Communications<>'None';

# Insert Agriculture & Natural Resources
INSERT INTO business_sector_activity (business_id, sector, activity)
SELECT id, 'Agriculture, Forestry & Natural Resources', AFN_Activities
FROM business_data
WHERE AFN_Activities IS NOT NULL
AND AFN_Activities<>'N/A'
AND AFN_Activities<>'None';

# Industrial Plants
INSERT INTO business_sector_activity (business_id, sector, activity)
SELECT id, 'Industrial Plants & Factories', Industrial_Factories
FROM business_data
WHERE Industrial_Factories IS NOT NULL
AND Industrial_Factories<>'N/A'
AND Industrial_Factories<>'None';


 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
5 rows affected.
7 rows affected.
5 rows affected.
0 rows affected.
0 rows affected.
0 rows affected.
5 rows affected.
0 rows affected.
1 rows affected.
1 rows affected.
3 rows affected.


""


In [65]:
%%sql
# Results
SELECT *
FROM business_sector_activity
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/business_data
5 rows affected.


,id,business_id,sector,activity
0,1,2,"General Traders, Shop & Retail Services",Hardware
1,2,3,"General Traders, Shop & Retail Services",Hardware
2,3,4,"General Traders, Shop & Retail Services",Boutique/Exhibition
3,4,5,"General Traders, Shop & Retail Services",None
4,5,6,"General Traders, Shop & Retail Services",Wines and Spirits


# EXPLORATORY DATA ANALYSIS(EDA)

### Survey Distribution by Area

Weight of each location in total dataset.

In [102]:
%%sql
CREATE OR REPLACE VIEW survey_distribution AS
SELECT 
    Place, 
    COUNT(*) AS surveys_completed,
    ROUND(COUNT(*) * 100.0 / (SELECT COUNT(*) FROM business_data), 2) AS percent_of_total_study
FROM business_data
GROUP BY Place
ORDER BY surveys_completed DESC;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.


""


In [103]:
%%sql
SELECT*
FROM survey_distribution;

 * mysql+pymysql://root:***@localhost:3306/business_data
3 rows affected.


,Place,surveys_completed,percent_of_total_study
0,Lower Marindi,33,50.00
1,Wasweta 1,30,45.45
2,Oruba,3,4.55


### General Business Count by Gender

In [112]:
%%sql
CREATE OR REPLACE VIEW general_business_gender_count AS
SELECT
    Gender, 
    COUNT(*) AS total_count,
    ROUND(COUNT(*) * 100.0 / (SELECT COUNT(*) FROM business_data), 2) AS percentage
FROM business_data
GROUP BY Gender;

SELECT*
FROM general_business_gender_count;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
2 rows affected.


,Gender,total_count,percentage
0,Male,26,39.39
1,Female,40,60.61


### General Business Count by Gender and Place

In [113]:
%%sql
CREATE OR REPLACE VIEW gender_business_count_place_gender AS
SELECT
    Place,
    Gender, 
    COUNT(*) AS total_count,
    ROUND(COUNT(*) * 100.0 / (SELECT COUNT(*) FROM business_data), 2) AS percentage
FROM business_data
GROUP BY Gender,Place;
SELECT*
FROM gender_business_count_place_gender;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
5 rows affected.


,Place,Gender,total_count,percentage
0,Lower Marindi,Male,11,16.67
1,Lower Marindi,Female,22,33.33
2,Wasweta 1,Female,18,27.27
3,Wasweta 1,Male,12,18.18
4,Oruba,Male,3,4.55


### Gender Distribution by Sector
Shows which industries are male-dominated vs. female-dominated.

In [115]:
%%sql
CREATE OR REPLACE VIEW gender_sector_analysis AS
SELECT 
    b.Gender,
    s.sector,
    COUNT(DISTINCT b.id) AS unique_businesses
FROM business_data b
JOIN business_sector_activity s ON b.id = s.business_id
WHERE s.activity <> 'None' AND s.activity <> 'N/A'
GROUP BY b.Gender, s.sector
ORDER BY s.sector, unique_businesses DESC;
SELECT*
FROM gender_sector_analysis;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
13 rows affected.


,Gender,sector,unique_businesses
0,Female,Accommodation & Catering,4
1,Male,Accommodation & Catering,3
2,Male,"Agriculture, Forestry & Natural Resources",1
3,Female,Communications,1
4,Female,"General Traders, Shop & Retail Services",25
5,Male,"General Traders, Shop & Retail Services",16
6,Female,Industrial Plants & Factories,3
7,Male,Informal Sector,3
8,Female,Informal Sector,2
9,Female,"Professional, Technical & Financial Services",3


### Identifying The Average Business Owner-Profiling

In [123]:
%%sql
CREATE OR REPLACE VIEW average_business_owner_profiling AS
SELECT 
    Place,
    (SELECT `Age Group` FROM business_data b2 WHERE b2.Place = b1.Place GROUP BY `Age Group` ORDER BY COUNT(*) DESC LIMIT 1) AS dominant_age_group,
    (SELECT Education FROM business_data b2 WHERE b2.Place = b1.Place GROUP BY Education ORDER BY COUNT(*) DESC LIMIT 1) AS dominant_education,
    COUNT(*) AS total_businesses
FROM business_data b1
GROUP BY Place;
SELECT* 
FROM average_business_owner_profiling;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
3 rows affected.


,Place,dominant_age_group,dominant_education,total_businesses
0,Lower Marindi,36 - 45 Yrs,Secondary,33
1,Wasweta 1,26 - 35 Yrs,Secondary,30
2,Oruba,26 - 35 Yrs,Secondary,3


### Place vs. Ownership Category

In [110]:
%%sql
CREATE OR REPLACE VIEW place_vs_ownership AS 
SELECT 
    Place,
    Ownership_Category,
    COUNT(*) AS count
FROM business_data
GROUP BY Place, Ownership_Category
ORDER BY Place, count DESC;
SELECT*
FROM place_vs_ownership;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
6 rows affected.


,Place,Ownership_Category,count
0,Lower Marindi,Sole Proprietor,21
1,Lower Marindi,Partnership,7
2,Lower Marindi,N/A,5
3,Oruba,Sole Proprietor,3
4,Wasweta 1,Sole Proprietor,18
5,Wasweta 1,N/A,12


### Analyzing Total Business Per sector

In [111]:
%%sql
CREATE OR REPLACE VIEW total_business_per_sector AS 
SELECT 
    sector, 
    COUNT(DISTINCT business_id) AS unique_businesses
FROM business_sector_activity
WHERE activity <> 'None'
GROUP BY sector
ORDER BY unique_businesses DESC;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.


""


In [96]:
%%sql
SELECT*
FROM total_business_per_sector;

 * mysql+pymysql://root:***@localhost:3306/business_data
8 rows affected.


,sector,unique_businesses
0,"General Traders, Shop & Retail Services",41
1,Accommodation & Catering,7
2,Informal Sector,5
3,"Professional, Technical & Financial Services",5
4,Transport,5
5,Industrial Plants & Factories,3
6,"Agriculture, Forestry & Natural Resources",1
7,Communications,1


### Business Categorizing Based On Employee Size

In [116]:
%%sql
CREATE OR REPLACE VIEW categorized_business_data AS
SELECT 
    *,
    CASE 
        WHEN Employee_Size = '1-5 employees' THEN 'Micro'
        WHEN Employee_Size = '6-10 employees' THEN 'Small'
        WHEN Employee_Size = 'Above 10' THEN 'Medium/Large'
        ELSE 'Not Specified'
    END AS Business_Category
FROM business_data;
SELECT Employee_Size, Business_Category
FROM categorized_business_data;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
66 rows affected.


,Employee_Size,Business_Category
0,1-5 employees,Micro
1,1-5 employees,Micro
2,1-5 employees,Micro
3,1-5 employees,Micro
4,1-5 employees,Micro
...,...,...
61,1-5 employees,Micro
62,1-5 employees,Micro
63,1-5 employees,Micro
64,1-5 employees,Micro


In [119]:
%%sql
CREATE OR REPLACE VIEW business_category AS 
SELECT Business_Category, COUNT(Business_Category) AS Unique_Business_Category
FROM categorized_business_data
GROUP BY Business_Category
ORDER BY Unique_Business_Category DESC;
SELECT*
FROM business_category;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
4 rows affected.


,Business_Category,Unique_Business_Category
0,Micro,54
1,Small,6
2,Medium/Large,4
3,Not Specified,2


In [118]:
%%sql
SELECT *
FROM categorized_business_data
WHERE Business_Category='Not Specified'

 * mysql+pymysql://root:***@localhost:3306/business_data
2 rows affected.


,Place,Gender,Age Group,Education,Role (Owner/Emp),Ownership_Category,Employee_Size,Years_Operated,retail_services,Informal_Sector,...,High_cost,Taxes,Acces_finance,Access_premises,Infrastructure_gaps,Sanitation,Competition,Other Issues,id,Business_Category
0,Wasweta 1,Female,Above 55 Yrs,University,Owner,Sole Proprietor,N/A,Less than 1 yr,Retail Shop,N/A,...,3,N/A,N/A,2,N/A,N/A,1,N/A,46,Not Specified
1,Lower Marindi,Female,36 - 45 Yrs,Primary,Owner,Sole Proprietor,N/A,6-10 yrs,N/A,N/A,...,1,2,N/A,N/A,3,N/A,N/A,N/A,57,Not Specified


### Gender vs. Business Category (Firm Size)
We can see if gender influences the size of the firm.

In [121]:
%%sql
SELECT 
    Business_Category,
    Gender,
    COUNT(*) AS total_businesses,
    ROUND(COUNT(*) * 100.0 / (SELECT COUNT(*) FROM business_data), 2) AS percentage_of_total
FROM categorized_business_data
WHERE Business_Category <> 'Not Specified'
GROUP BY Business_Category, Gender
ORDER BY FIELD(Business_Category, 'Micro', 'Small', 'Medium/Large'), Gender;

 * mysql+pymysql://root:***@localhost:3306/business_data
6 rows affected.


,Business_Category,Gender,total_businesses,percentage_of_total
0,Micro,Female,33,50.00
1,Micro,Male,21,31.82
2,Small,Female,3,4.55
3,Small,Male,3,4.55
4,Medium/Large,Female,2,3.03
5,Medium/Large,Male,2,3.03


### Micro Concentration
**81.82% (50% Female + 31.82% Male)** of total businesses are Micro (1–5 employees). This suggests that the local economy is dominated by small-scale, likely family-run or individual enterprises rather than industrial or large-scale employers.

#### Key Observations
1. **The Gender Scaling Gap**-While women represent the highest percentage of business owners overall, they are overwhelmingly concentrated in the Micro category (33 out of 38 female-led businesses).
2. The numbers drop sharply after the Micro level. Only 6 businesses total are "Small," and only 4 are "Medium/Large."
3. **Equality at the Top**. Small and Medium/Large levels, the numbers are perfectly split between Male and Female (3 each for Small, 2 each for Medium). This suggests that while fewer women scale their businesses, those who do are just as capable of reaching the higher tiers as men.

### Business Maturity by Place (Survival Analysis)
Are businesses in Lower Marindi newer compared to Wasweta 1 or Oruba?

In [126]:
%%sql
SELECT 
    Place,
    Years_Operated,
    COUNT(*) AS business_count
FROM business_data
GROUP BY Place, Years_Operated
ORDER BY Place, field(Years_Operated, '1-5 yrs', '6-10 yrs', 'Above 10');

 * mysql+pymysql://root:***@localhost:3306/business_data
12 rows affected.


,Place,Years_Operated,business_count
0,Lower Marindi,Less than 1 yr,6
1,Lower Marindi,More than 10 yrs,3
2,Lower Marindi,1-5 yrs,15
3,Lower Marindi,6-10 yrs,9
4,Oruba,More than 10 yrs,1
5,Oruba,1-5 yrs,1
6,Oruba,6-10 yrs,1
7,Wasweta 1,Less than 1 yr,5
8,Wasweta 1,More than 10 yrs,4
9,Wasweta 1,N/A,1


### Lower Marindi:Startup Hub
Lower Marindi has the highest number of brand-new businesses (6 businesses operated for "Less than 1 yr").
1. It has a very high concentration of young businesses (21 total under 5 years).
It is "newer" in terms of recent entries, suggesting it might be a growing commercial node or has lower barriers to entry for new entrepreneurs.

### Wasweta 1: Balanced Area
1. Wasweta 1 also shows healthy new growth (5 businesses under 1 yr) and matches Lower Marindi exactly in the 1-5 yrs bracket (15 businesses).
It has a similar "youthful" business profile to Lower Marindi but with slightly fewer very old businesses.

### Oruba: Survival
1. Oruba is the outlier. It has zero businesses reported in the "Less than 1 yr" category and only 1 in the "1-5 yrs" category.
Almost its entire sample is composed of older businesses (6-10 years or More than 10 years).

Businesses in Oruba are technically "older" (more mature). Still, the absence of new businesses (0 under 1 year) could indicate a saturated market, higher entry costs, or a declining business environment relative to the other two areas.

### NOTE: Oruba has a very small sample size in this output only 3 records which might skew the comparison

### Productivity: Employee Size vs. Gender
This checks if male-led or female-led businesses tend to hire more people, Employee_Size.

In [129]:
%%sql
CREATE OR REPLACE VIEW productivity AS
SELECT 
    Gender,
    Employee_Size,
    COUNT(*) AS count
FROM business_data
GROUP BY Gender, Employee_Size
ORDER BY Gender, count DESC;
SELECT*
FROM productivity;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
7 rows affected.


,Gender,Employee_Size,count
0,Female,1-5 employees,33
1,Female,6-10 employees,3
2,Female,Above 10,2
3,Female,N/A,2
4,Male,1-5 employees,21
5,Male,6-10 employees,3
6,Male,Above 10,2


1. **Micro-Entrepreneurs Majority**: Both genders are heavily concentrated in the 1-5 employees bracket. However, the concentration is much higher for women (33 businesses) compared to men (21 businesses).

2. **Scaling to Small(6-10)**: Both genders have the same number of businesses that have scaled to the 6-10 employee range (3 each). This shows that once a business survives the initial startup phase, gender becomes less of a factor in hiring up to 10 people.

3. **Medium/Large Tier (Above 10)**: Both Male and Female-led businesses have 2 units each in the "Above 10" category.

### Hiring Capacity by Gender
If we calculate the Expansion Rate (percentage of businesses that grow beyond 5 employees):

   1. Women: Only about 12.5% of female-led businesses hire more than 5 people.
   2. Men: About 19.2% of male-led businesses hire more than 5 people.

This suggests that while women are more likely to start a business in these areas, male-led businesses in our current sample have a slightly higher tendency to hire more staff. This could be due to the Sector (e.g., if men are in Industrial/Transport, which requires more labor) or Access to Finance.

### HIRING CAPACITY CALCULATION
Total Female Businesses: 33(Micro)+ 3(Small)+ 2(Medium/Large)+ 2(N/A) =40
Total Male Businesses:21(Micro)+ 3(Small)+ 2(Medium/Large)=26

#### Identifying the Expansion Group.
The Expansion Rate considers any business that has grown beyond the Micro (1–5) stage. That means we sum the "6–10" and "Above 10"  
categories:  
Female Expanders:3(Small) + 2(Medium/Large)=5  
Male Expanders:3(Small)+ 2(Medium/Large)=5

Expansion Rate=Expanders/Total*100
##### Female
5/40=0.125*100
=12.5%

##### Male
5/26=0.1923*100
=19.2%

Even though both genders have the exact same number of expanded businesses (5 each), the rate is different because there are so many more women starting Micro-businesses.This indicates:
1. Female Entrepreneurship is a massive driver for self-employment.
2. Male Entrepreneurship in this specific sample has a slightly higher "Scaling Velocity."

### Growth Stagnation
Across all areas, the vast majority of businesses (roughly 85%) are not expanding beyond 5 employees. This Micro-Trap along the A1 road suggests that while the road provides high visibility, businesses may be struggling with high operating costs or limited access to expansion capital.

In [130]:
%%sql
CREATE OR REPLACE VIEW  gender_business_expansion_rate AS 
SELECT 
    Gender,
    COUNT(*) AS total_businesses,
    SUM(CASE WHEN Employee_Size IN ('6-10 employees', 'Above 10') THEN 1 ELSE 0 END) AS expanders,
    ROUND(
        (SUM(CASE WHEN Employee_Size IN ('6-10 employees', 'Above 10') THEN 1 ELSE 0 END) / COUNT(*)) * 100, 
        2
    ) AS expansion_rate_percent
FROM business_data
WHERE Employee_Size <> 'N/A' -- Optional: removes outliers for a cleaner rate
GROUP BY Gender;

SELECT*
FROM gender_business_expansion_rate;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
2 rows affected.


,Gender,total_businesses,expanders,expansion_rate_percent
0,Male,26,5,19.23
1,Female,38,5,13.16


### Expansion Rate In Oruba, Lower Marindi,Wasweta 1

In [131]:
%%sql 
CREATE OR REPLACE VIEW location_business_expansion_rate AS
SELECT 
    Place,
    COUNT(*) AS total_surveys,
    SUM(CASE WHEN Employee_Size IN ('6-10 employees', 'Above 10', '6-10', 'Above 10') THEN 1 ELSE 0 END) AS expanded_businesses,
    ROUND(
        (SUM(CASE WHEN Employee_Size IN ('6-10 employees', 'Above 10', '6-10', 'Above 10') THEN 1 ELSE 0 END) / COUNT(*)) * 100, 
        2
    ) AS expansion_rate_percent
FROM business_data
GROUP BY Place
ORDER BY expansion_rate_percent DESC;
SELECT*
FROM location_business_expansion_rate;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
3 rows affected.


,Place,total_surveys,expanded_businesses,expansion_rate_percent
0,Lower Marindi,33,7,21.21
1,Wasweta 1,30,3,10.00
2,Oruba,3,0,0.00


#### Lower Marindi's Dominance
With a 21.21% expansion rate, Lower Marindi is twice as productive in terms of job creation as Wasweta 1. This suggests that the businesses located at this specific node of the A1 road have better access to customers, larger premises, or represent more mature sectors like wholesale or service centers.

### Wasweta 1's Micro Trap
Despite having a nearly identical total survey count (30 vs 33), only 3 businesses in Wasweta 1 have expanded. This area is heavily defined by Micro-enterprises (1-5 employees). 
#### The question then is, does Wasweta 1 have higher land fragmentation or smaller shop sizes that prevent physical expansion? 

### Oruba Limitation
As we noted, the 0.00% expansion rate in Oruba is not necessarily a sign of poor business performance, but a reflection of the small sample size (N=3). Since the survey focused on the A1 road, it implies that the specific segments nearing the road have a lower density of large-scale commercial activity compared to the other two nodes.

### Human Capital & Demographics Comparison
A comparative analysis between Lower Marindi and Wasweta 1 to see if the business owners in the faster-expanding area (Lower Marindi) have a different demographic profile.

In [133]:
%%sql
CREATE OR REPLACE VIEW human_capital_demo_comparison AS
SELECT 
    Place,
    Education,
    `Age Group`,
    COUNT(*) AS total_count,
    SUM(CASE WHEN Employee_Size IN ('6-10 employees', 'Above 10', '6-10', 'Above 10') THEN 1 ELSE 0 END) AS expanded_count
FROM business_data
WHERE Place IN ('Lower Marindi', 'Wasweta 1')
GROUP BY Place, Education, `Age Group`
ORDER BY Place, total_count DESC;
SELECT* 
FROM human_capital_demo_comparison;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
25 rows affected.


,Place,Education,Age Group,total_count,expanded_count
0,Lower Marindi,Secondary,26 - 35 Yrs,7,2
1,Lower Marindi,Secondary,18 - 25 Yrs,4,2
2,Lower Marindi,Secondary,36 - 45 Yrs,4,0
3,Lower Marindi,College,36 - 45 Yrs,4,2
4,Lower Marindi,College,26 - 35 Yrs,3,0
5,Lower Marindi,No Formal Education,36 - 45 Yrs,2,0
6,Lower Marindi,University,36 - 45 Yrs,2,1
7,Lower Marindi,College,18 - 25 Yrs,2,0
8,Lower Marindi,Primary,36 - 45 Yrs,2,0
9,Lower Marindi,Primary,26 - 35 Yrs,1,0


### CBD Effect: Why is Lower Marindi Scaling
Looking at the data, there’s a massive gap between Lower Marindi and Wasweta 1 that can’t be explained by education alone. We have owners with the exact same Secondary schooling and age profiles in both places, yet the ones in **Lower Marindi are scaling while the ones in Wasweta 1 are stuck** at the "Micro" level.

#### This is a textbook case of Agglomeration Economies that we learned. 
Lower Marindi isn’t just a location; it’s the heart of the CBD. Because it’s so vibrant and dense:
1. **Foot Traffic is a Constant**: A business here doesn't have to hunt for customers; the A1 road delivers them to the doorstep.
2. **Knowledge Spillovers**: Owners here are likely learning from each other, sharing suppliers, or spotting market trends faster because they are in the thick of it.
3. **Lower Transaction Costs**: Being in the hub means it’s easier to find casual labor, restock inventory, or access services.

### My Takeaways for the Thesis:
It’s not just who the entrepreneur is (Education/Age), but where they are. The "Expansion Rate" in Lower Marindi (21%) vs. Wasweta 1 (10%) proves that the CBD provides a "productivity bonus." If we want Wasweta 1 to grow, we shouldn't just focus on more training; we need to look at how to replicate the CBD's density better market infrastructure, better lighting, and better transport links to pull people in.

### Spatial Analysis of Ownership Structures
To determine if the vibrancy of Lower Marindi (CBD) encourages collaborative ownership (Partnerships) compared to the individualistic nature of Wasweta 1.

In [135]:
%%sql
CREATE OR REPLACE VIEW ownership_spatial_analysis AS
SELECT 
    Place,
    Ownership_Category,
    COUNT(*) AS total_count,
    -- Checking how many of these actually managed to scale
    SUM(CASE WHEN Employee_Size IN ('6-10 employees', 'Above 10', '6-10', 'Above 10') THEN 1 ELSE 0 END) AS expanded_count,
    ROUND(COUNT(*) * 100.0 / (SELECT COUNT(*) FROM business_data WHERE Place = b.Place), 2) AS percentage_within_place
FROM business_data b
WHERE Place IN ('Lower Marindi', 'Wasweta 1')
GROUP BY Place, Ownership_Category
ORDER BY Place, total_count DESC;
SELECT*
FROM ownership_spatial_analysis;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
5 rows affected.


,Place,Ownership_Category,total_count,expanded_count,percentage_within_place
0,Lower Marindi,Sole Proprietor,21,4,63.64
1,Lower Marindi,Partnership,7,0,21.21
2,Lower Marindi,N/A,5,3,15.15
3,Wasweta 1,Sole Proprietor,18,2,60.00
4,Wasweta 1,N/A,12,1,40.00


### The Survivalist vs Formal Split
**Even though Sole Proprietors are the majority everywhere**, the quality of that ownership varies wildly between the two areas:

#### The Hidden Formalization in the CBD: 
In Lower Marindi, over 21% of businesses have moved into Partnerships. **Even more interesting is the "N/A" category in the CBD, 3 out of 5 of those "unlabeled" businesses are actually the ones expanding**. This suggests that businesses in the heart of town are either too busy scaling to worry about labels, or they are operating under more complex structures (like family-run firms) that the survey didn't fully capture.

#### The Trap in Wasweta 1:
Wasweta 1 has zero recorded Partnerships. It is 100% split between Sole Proprietors and "N/A." This tells us that in Wasweta 1, the economy is much more individualistic. People are "going it alone."

**In the CBD (Lower Marindi), the high cost of rent and the high volume of customers almost demand more hands on deck. You see more people teaming up.** In Wasweta 1, because it's further from the "vibrant heart," the businesses remain smaller and more isolated.

### Summary
If you want to scale in Migori, you're better off being a 'Sole Proprietor' in the CBD than anywhere else. In Lower Marindi, even the solo guys are expanding (4 of them), likely because they are benefiting from the foot traffic of the big shops nearby. Meanwhile, in **Wasweta 1, 40% of the businesses are in that 'N/A' zone, which hints at informal, survivalist trading** that just doesn't have the guts to grow into a large employer.

### Sector Specialization & The Expansion Gap
To identify if the CBD Productivity Bonus in Lower Marindi is driven by higher-value sectors (Services/Wholesale) compared to the retail-heavy outskirts of Wasweta 1.

In [139]:
%%sql
CREATE OR REPLACE VIEW sector_specialization_expansion AS
SELECT 
    b.Place,
    s.sector,
    COUNT(DISTINCT b.id) AS unique_businesses,
    # Count how many of these unique businesses scaled
    COUNT(DISTINCT CASE WHEN b.Employee_Size IN ('6-10 employees', 'Above 10', '6-10', 'Above 10') THEN b.id END) AS expanded_count,
    # Percentage of the unique 66-record sample
    ROUND(COUNT(DISTINCT b.id) * 100.0 / (SELECT COUNT(DISTINCT id) FROM business_data WHERE Place = b.Place), 2) AS percent_of_local_economy
FROM business_data b
JOIN business_sector_activity s ON b.id = s.business_id
WHERE b.Place IN ('Lower Marindi', 'Wasweta 1')
  AND s.sector <> 'None'
GROUP BY b.Place, s.sector
ORDER BY b.Place, unique_businesses DESC;
SELECT*
FROM sector_specialization_expansion;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
14 rows affected.


,Place,sector,unique_businesses,expanded_count,percent_of_local_economy
0,Lower Marindi,"General Traders, Shop & Retail Services",18,2,54.55
1,Lower Marindi,Informal Sector,6,1,18.18
2,Lower Marindi,Accommodation & Catering,5,1,15.15
3,Lower Marindi,Transport,3,2,9.09
4,Lower Marindi,Industrial Plants & Factories,2,1,6.06
5,Lower Marindi,"Professional, Technical & Financial Services",2,0,6.06
6,Lower Marindi,"Agriculture, Forestry & Natural Resources",1,0,3.03
7,Lower Marindi,Communications,1,1,3.03
8,Wasweta 1,"General Traders, Shop & Retail Services",22,3,73.33
9,Wasweta 1,"Professional, Technical & Financial Services",3,0,10.00


In [142]:
%%sql
CREATE OR REPLACE VIEW sector_specialization_aggregation AS
SELECT 
    Place,
    CASE 
        WHEN sector LIKE '%General Traders%' THEN 'General Traders'
        WHEN sector LIKE '%Informal Sector%' THEN 'Informal Sector'
        WHEN sector LIKE '%Professional%' THEN 'Professional/Financial'
        ELSE 'Other Specialized'
    END AS Sector_Grouped,
    SUM(unique_businesses) AS Total_Businesses,
    SUM(expanded_count) AS Total_Expanded,
    ROUND(AVG(percent_of_local_economy), 1) AS Avg_Market_Share_Percent
FROM sector_specialization_expansion
WHERE Place IN ('Lower Marindi', 'Wasweta 1')
GROUP BY Place, Sector_Grouped
ORDER BY Place, Total_Businesses DESC;
SELECT* 
FROM sector_specialization_aggregation;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
8 rows affected.


,Place,Sector_Grouped,Total_Businesses,Total_Expanded,Avg_Market_Share_Percent
0,Lower Marindi,General Traders,18,2,54.6
1,Lower Marindi,Other Specialized,12,5,7.3
2,Lower Marindi,Informal Sector,6,1,18.2
3,Lower Marindi,Professional/Financial,2,0,6.1
4,Wasweta 1,General Traders,22,3,73.3
5,Wasweta 1,Other Specialized,4,0,4.4
6,Wasweta 1,Professional/Financial,3,0,10.0
7,Wasweta 1,Informal Sector,2,0,6.7


### Specialized Engine vs. The Retail
#### Quality over Quantity: The Other Specialized Factor
In planning, we often look for the **economic engine** of a town. The data shows that while Retail is more common, **the Other Specialized category (Industrial, Transport, Communications) is the real job creator in the CBD.**

**Lower Marindi**: Out of 12 specialized businesses, 5 have expanded. That is a 41.7% scaling rate.

**Wasweta 1**: Out of 4 specialized businesses, 0 have expanded. That is a 0% scaling rate.

In the CBD, a specialized business is nearly 42% likely to expand, whereas in the periphery, these same sectors struggle to break the 5-employee ceiling. Location acts as a catalyst for high-order services.

#### The Retail Monopoly(Wasweta 1)
Wasweta 1 is essentially "locked" into General Trading, which accounts for 73.3% of its local economy. This is a **sectoral over-specialization**.
Because almost everyone is selling similar goods to the same limited local market, the "Expansion Velocity" is incredibly low.

Even though Wasweta 1 has 22 traders (more than Lower Marindi’s 18), only 3 have managed to scale. This represents a modest 13.6% expansion rate for retail in the outskirts compared to the high-growth specialized sectors in the heart of town.

### Vibrancy = Diversity (The Agglomeration Economy)
The vibrancy of Migori’s CBD is more than just a feeling; it’s visible in the "Other Specialized" count.

**Lower Marindi has 3x more specialized businesses (Industrial, Transport, Communication) than Wasweta 1.**

This diversity is the foundation of an Agglomeration Economy. In the CBD, businesses don't just exist side by side; they support each other. A transport firm helps a wholesaler, who in turn hires a professional service. This synergy is what allows these businesses to grow, whereas the "Retail Monopoly" in Wasweta 1 leads to stagnation.

In [141]:
%sql DESCRIBE business_data;

 * mysql+pymysql://root:***@localhost:3306/business_data
29 rows affected.


,Field,Type,Null,Key,Default,Extra
0,Place,text,YES,,None,
1,Gender,text,YES,,None,
2,Age Group,text,YES,,None,
3,Education,text,YES,,None,
4,Role (Owner/Emp),text,YES,,None,
5,Ownership_Category,text,YES,,None,
6,Employee_Size,text,YES,,None,
7,Years_Operated,text,YES,,None,
8,retail_services,text,YES,,None,
9,Informal_Sector,text,YES,,None,


### Education-Gender Gap
Understand if women in certain areas have different educational backgrounds than men

In [143]:
%%sql
CREATE OR REPLACE VIEW education_gender_gap AS
SELECT 
    Place,
    Gender,
    Education,
    COUNT(*) AS count
FROM business_data
GROUP BY Place, Gender, Education
ORDER BY Place, count DESC;
SELECT*
FROM education_gender_gap;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
18 rows affected.


,Place,Gender,Education,count
0,Lower Marindi,Female,Secondary,9
1,Lower Marindi,Female,College,7
2,Lower Marindi,Male,Secondary,6
3,Lower Marindi,Male,College,3
4,Lower Marindi,Female,No Formal Education,2
5,Lower Marindi,Male,Primary,2
6,Lower Marindi,Female,Primary,2
7,Lower Marindi,Female,University,2
8,Oruba,Male,Secondary,2
9,Oruba,Male,Primary,1


Why are these educated women stuck in the **Retail Trap** while the CBD’s **Specialized** sector scales?

Access vs. Tenure. Beyond just the hurdles, we need to see how long these businesses have been running and if the women are getting access to the same resources.

### The Gendered Barrier Analysis
To see if educated women are citing **Finance** and **Premises** as their hurdle more often than men, which would explain the Retail Monopoly Trap.
Less than 1yr -> 0.5  
1-5 yrs ->3  
6-10 yrs->8  
More than 10 yrs->12 (Conservative estimate)

In [9]:
%%sql 
CREATE OR REPLACE VIEW gender_barrier_summary AS
SELECT 
    Place,
    Gender,
    Education,
    ROUND(AVG(CASE 
        WHEN Years_Operated = 'Less than 1 yr' THEN 0.5
        WHEN Years_Operated = '1-5 yrs' THEN 3
        WHEN Years_Operated = '6-10 yrs' THEN 8
        WHEN Years_Operated = 'More than 10 yrs' THEN 12
        ELSE 0 
    END), 1) AS avg_years_tenure,
    # Top Hurdle Counts for ALL 7 Categories
    SUM(CASE WHEN High_cost = '1' THEN 1 ELSE 0 END) AS hurdle_high_cost,
    SUM(CASE WHEN Taxes = '1' THEN 1 ELSE 0 END) AS hurdle_taxes,
    SUM(CASE WHEN Acces_finance = '1' THEN 1 ELSE 0 END) AS hurdle_finance,
    SUM(CASE WHEN Access_premises = '1' THEN 1 ELSE 0 END) AS hurdle_premises,
    SUM(CASE WHEN Infrastructure_gaps = '1' THEN 1 ELSE 0 END) AS hurdle_infra,
    SUM(CASE WHEN Sanitation = '1' THEN 1 ELSE 0 END) AS hurdle_sanitation,
    SUM(CASE WHEN Competition = '1' THEN 1 ELSE 0 END) AS hurdle_competition,
    # Scaling success
    SUM(CASE WHEN Employee_Size IN ('6-10 employees', 'Above 10', '6-10', 'Above 10') THEN 1 ELSE 0 END) AS expansion_count
FROM business_data
GROUP BY Place, Gender, Education;

SELECT * FROM gender_barrier_summary;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
18 rows affected.


,Place,Gender,Education,avg_years_tenure,hurdle_high_cost,hurdle_taxes,hurdle_finance,hurdle_premises,hurdle_infra,hurdle_sanitation,hurdle_competition,expansion_count
0,Lower Marindi,Male,Secondary,4.2,2,1,1,0,0,1,0,3
1,Lower Marindi,Female,No Formal Education,3.0,0,1,1,0,0,0,0,0
2,Lower Marindi,Female,Secondary,4.1,2,3,1,1,1,1,1,1
3,Lower Marindi,Female,College,4.7,4,1,0,0,1,1,0,2
4,Wasweta 1,Female,No Formal Education,3.0,1,0,0,0,0,0,0,0
5,Wasweta 1,Female,Secondary,4.6,2,0,1,0,0,1,2,0
6,Wasweta 1,Male,College,5.2,1,0,1,0,0,0,2,0
7,Lower Marindi,Male,Primary,3.0,0,2,0,0,0,0,0,0
8,Lower Marindi,Female,Primary,8.0,1,0,0,0,0,1,0,0
9,Wasweta 1,Male,Secondary,4.4,2,0,0,0,0,1,2,1


### Stress Profile
#### The CBD's(Lower Marindi)
Row 2 (Female, Secondary) and row 3 (Female, College) in Lower Marindi:  
1. The Dominant Hurdle: High Cost and Taxes.
These **educated women are the ones scaling (3 expansions combined)**, but they are being squeezed by the **Formal Economy** (Rent + Taxes).  

In the CBD, growth isn't limited by a lack of customers, but by the cost of entry. To make the CBD more **vibrant**, the policy should focus on tax holidays or rent control.

### The Periphery's Market Saturation(Wasweta 1)
Row 5(Female, Secondary) and Row 12(Female, College) in Wasweta 1
1. The Dominant Hurdle: Competition (4 total) and High Cost (5 total).  
Even with nearly 5 years of experience, these women are struggling to expand.

On the outskirts, the problem is **Copycat Economics**. **Everyone is selling the same thing**, leading to **high competition** and **low profits**. Education doesn't save you here because the market is simply too crowded.

### The Invisible Gender Burden (Sanitation & Infra)
The hurdle_sanitation and hurdle_infra columns. These rarely appear for men as a no.1  hurdle.  
For Female Secondary/College owners in both Marindi and Wasweta, these issues are starting to pop up as top priorities.

Poor lighting (Infra) and waste management (Sanitation) act as a time poverty tax on women, limiting their business hours and growth.

#### The Oruba (Rows 16 & 17)
Oruba is looking very different.  
We have a Male with 12 years of experience (Primary education) who cites Taxes as the only hurdle, but has 0 expansions.

This suggests Oruba might not be a **growth node** at all, but a **Survival Node** where people stay in business for a long time without ever getting bigger.



### Ownership Barrier Analysis
To determine if the structure of a business (Sole vs. Partnership) changes the type of obstacles it faces.

In [10]:
%%sql
CREATE OR REPLACE VIEW ownership_barrier_analysis AS
SELECT 
    Ownership_Category,
    Place,
    COUNT(*) AS total_businesses,
    # Top Hurdle Counts
    SUM(CASE WHEN High_cost = '1' THEN 1 ELSE 0 END) AS hurdle_high_cost,
    SUM(CASE WHEN Taxes = '1' THEN 1 ELSE 0 END) AS hurdle_taxes,
    SUM(CASE WHEN Acces_finance = '1' THEN 1 ELSE 0 END) AS hurdle_finance,
    SUM(CASE WHEN Access_premises = '1' THEN 1 ELSE 0 END) AS hurdle_premises,
    SUM(CASE WHEN Infrastructure_gaps = '1' THEN 1 ELSE 0 END) AS hurdle_infra,
    SUM(CASE WHEN Sanitation = '1' THEN 1 ELSE 0 END) AS hurdle_sanitation,
    SUM(CASE WHEN Competition = '1' THEN 1 ELSE 0 END) AS hurdle_competition,
    # Expansion Performance
    SUM(CASE WHEN Employee_Size IN ('6-10 employees', 'Above 10', '6-10', 'Above 10') THEN 1 ELSE 0 END) AS expansion_count
FROM business_data
WHERE Ownership_Category IS NOT NULL
GROUP BY Ownership_Category, Place;

SELECT * 
FROM ownership_barrier_analysis;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
6 rows affected.


,Ownership_Category,Place,total_businesses,hurdle_high_cost,hurdle_taxes,hurdle_finance,hurdle_premises,hurdle_infra,hurdle_sanitation,hurdle_competition,expansion_count
0,Partnership,Lower Marindi,7,2,2,2,0,0,1,0,0
1,Sole Proprietor,Lower Marindi,21,7,7,1,0,2,3,0,4
2,Sole Proprietor,Wasweta 1,18,8,1,1,0,0,1,5,2
3,N/A,Wasweta 1,12,3,0,1,0,0,2,6,1
4,N/A,Lower Marindi,5,2,1,0,1,0,1,1,3
5,Sole Proprietor,Oruba,3,0,3,0,0,0,0,0,0


### Partnerships
Row 0(Partnership, Lower Marindi)
Out of 7 Partnerships, all 7 cited a top hurdle, with High Cost and Taxes being the primary complaints (2 each).

**Despite being a more formal structure, only 0 expanded in this group.**
Usually, partnerships are the vehicles for growth. In the CBD, it seems that once you form a partnership and become visible, the high costs of rent and the tax burden neutralize your ability to hire more people.

### Sole Proprietor (Wasweta 1 & Lower Marindi)
Row 1 & 2
In Wasweta 1, Sole Proprietors cite Competition (5) as their no.1 killer. In the CBD, they cite High Cost (7).

Surprisingly, Sole Proprietors are doing the heavy lifting. We have 4 expansions in Lower Marindi and 2 in Wasweta 1 coming from individuals.

Migori’s economy is currently a **One-Man/Woman Show**. The job growth isn't coming from organized firms, but from highly motivated individuals who are scaling up despite the chaos.

### The Employee Perspective (N/A Rows 3 & 4)
The N/A entries represent businesses where employees were interviewed, providing a front-line validation of Migori’s economic growth that differs from owner-reported data.  
In Lower Marindi, this group shows a high expansion rate (3 out of 5 businesses), but they are the only respondents to rank Sanitation and Infrastructure Gaps as their no.1 1hurdles. 

This suggests that while these businesses are successful enough to hire staff, the physical urban environment is failing the actual workers who spend their days on-site, shifting the planning focus from financial relief for owners to basic service delivery for the labor force.

### Sectoral Hurdle Analysis
To determine if high-growth specialized sectors face different physical or regulatory barriers than the retail sector.

In [13]:
%%sql
CREATE OR REPLACE VIEW sector_hurdle_analysis_v3 AS
SELECT 
    CASE 
        WHEN retail_services IS NOT NULL AND retail_services != 'N/A' AND retail_services != '' THEN 'Retail & General Trade'
        WHEN Industrial_Factories IS NOT NULL AND Industrial_Factories != 'N/A' AND Industrial_Factories != '' THEN 'Industrial/Manufacturing'
        WHEN (Transport IS NOT NULL AND Transport != 'N/A' AND Transport != '') 
          OR (Storage IS NOT NULL AND Storage != 'N/A' AND Storage != '') 
          OR (Communications IS NOT NULL AND Communications != 'N/A' AND Communications != '') THEN 'Logistics & Tech'
        WHEN Informal_Sector IS NOT NULL AND Informal_Sector != 'N/A' AND Informal_Sector != '' THEN 'Informal/Small Scale'
        WHEN (Accomodation_Catering IS NOT NULL AND Accomodation_Catering != 'N/A' AND Accomodation_Catering != '') 
          OR (Entertainment IS NOT NULL AND Entertainment != 'N/A' AND Entertainment != '') THEN 'Hospitality & Leisure'
        WHEN (PTF_Services IS NOT NULL AND PTF_Services != 'N/A' AND PTF_Services != '') 
          OR (Health IS NOT NULL AND Health != 'N/A' AND Health != '') 
          OR (Education_Institution IS NOT NULL AND Education_Institution != 'N/A' AND Education_Institution != '') THEN 'Professional & Social Services'
        WHEN AFN_Activities IS NOT NULL AND AFN_Activities != 'N/A' AND AFN_Activities != '' THEN 'Primary Sector (Agri/Forest/Res)'
        ELSE 'Unclassified'
    END AS Detailed_Sector_Group,
    Place,
    COUNT(*) AS total_businesses,
    # Top Hurdle Counts
    SUM(CASE WHEN High_cost = '1' THEN 1 ELSE 0 END) AS hurdle_high_cost,
    SUM(CASE WHEN Taxes = '1' THEN 1 ELSE 0 END) AS hurdle_taxes,
    SUM(CASE WHEN Acces_finance = '1' THEN 1 ELSE 0 END) AS hurdle_finance,
    SUM(CASE WHEN Access_premises = '1' THEN 1 ELSE 0 END) AS hurdle_premises,
    SUM(CASE WHEN Infrastructure_gaps = '1' THEN 1 ELSE 0 END) AS hurdle_infra,
    SUM(CASE WHEN Sanitation = '1' THEN 1 ELSE 0 END) AS hurdle_sanitation,
    SUM(CASE WHEN Competition = '1' THEN 1 ELSE 0 END) AS hurdle_competition,
    # Scaling success
    SUM(CASE WHEN Employee_Size IN ('6-10 employees', 'Above 10', '6-10', 'Above 10') THEN 1 ELSE 0 END) AS expansion_count
FROM business_data
GROUP BY Detailed_Sector_Group, Place;

SELECT * FROM sector_hurdle_analysis_v3;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
16 rows affected.


,Detailed_Sector_Group,Place,total_businesses,hurdle_high_cost,hurdle_taxes,hurdle_finance,hurdle_premises,hurdle_infra,hurdle_sanitation,hurdle_competition,expansion_count
0,Informal/Small Scale,Lower Marindi,3,1,1,1,0,0,0,0,0
1,Retail & General Trade,Lower Marindi,17,6,7,2,0,0,2,0,2
2,Hospitality & Leisure,Lower Marindi,4,1,1,0,0,0,1,0,1
3,Retail & General Trade,Wasweta 1,22,8,1,1,0,0,3,7,3
4,Industrial/Manufacturing,Lower Marindi,2,1,0,0,0,0,1,0,1
5,Logistics & Tech,Lower Marindi,4,2,1,0,0,1,0,0,3
6,Hospitality & Leisure,Wasweta 1,1,1,0,0,0,0,0,0,0
7,Professional & Social Services,Wasweta 1,2,1,0,1,0,0,0,0,0
8,Logistics & Tech,Wasweta 1,2,0,0,0,0,0,0,2,0
9,Informal/Small Scale,Wasweta 1,2,0,0,0,0,0,0,2,0


### Sectoral Barrier Analysis: Growth vs. Stagnation
1. The High-Growth Specialized Sectors (Logistics, Tech, & Manufacturing)
The Logistics & Tech group in Lower Marindi is the **Economic Engine** with a 75% expansion rate (3 out of 4).

Unlike Retail, these specialized businesses are the only ones citing Infrastructure Gaps (1) and High Costs (2) as their primary friction points.
These sectors are ready to scale, but they are physically limited. Their growth depends on the county’s ability to provide stable **Backend** infrastructure (power, roads, and digital connectivity).

3. The Low-Growth Retail
In Wasweta 1, Retail & General Trade is the largest employer (22 businesses) but has a low expansion rate of only 13%.  
This sector is uniquely crushed by Competition (7) and Sanitation (3).  

Retail is in a Survival Loop. Because there are too many identical shops in the same space, profits are thin, and expansion is nearly impossible. Planning for Wasweta 1 shouldn't just be about more space for shops, but about zoning for variety to reduce this cut-throat competition.

3. The Institutional Burden (Taxes)
Taxes are almost exclusively a CBD problem. In Lower Marindi, Retail (7) and Logistics (1) cite taxes heavily. In Wasweta 1, taxes almost disappear as a no.1 hurdle.

There is a **Formalization Penalty** in the CBD. Businesses that grow large enough to be noticed by the county are being taxed at a rate that may be hindering their next stage of hiring.

### Other Issues Analysis

In [27]:
%%sql
CREATE OR REPLACE VIEW business_stagnation_voices AS
SELECT 
    id, -- Added ID for easier reference
    Place,
    Detailed_Sector_Group,
    `Other Issues`,
    Access_premises
FROM (
    SELECT 
        id,
        Place,
        CASE 
            WHEN retail_services IS NOT NULL AND retail_services NOT IN ('', 'N/A', '0') THEN 'Retail & General Trade'
            WHEN Industrial_Factories IS NOT NULL AND Industrial_Factories NOT IN ('', 'N/A', '0') THEN 'Industrial/Manufacturing'
            WHEN (Transport IS NOT NULL AND Transport NOT IN ('', 'N/A', '0')) 
              OR (Storage IS NOT NULL AND Storage NOT IN ('', 'N/A', '0')) 
              OR (Communications IS NOT NULL AND Communications NOT IN ('', 'N/A', '0')) THEN 'Logistics & Tech'
            WHEN Informal_Sector IS NOT NULL AND Informal_Sector NOT IN ('', 'N/A', '0') THEN 'Informal/Small Scale'
            WHEN (Accomodation_Catering IS NOT NULL AND Accomodation_Catering NOT IN ('', 'N/A', '0')) 
              OR (Entertainment IS NOT NULL AND Entertainment NOT IN ('', 'N/A', '0')) THEN 'Hospitality & Leisure'
            WHEN (PTF_Services IS NOT NULL AND PTF_Services NOT IN ('', 'N/A', '0')) 
              OR (Health IS NOT NULL AND Health NOT IN ('', 'N/A', '0')) 
              OR (Education_Institution IS NOT NULL AND Education_Institution NOT IN ('', 'N/A', '0')) THEN 'Professional & Social Services'
            WHEN AFN_Activities IS NOT NULL AND AFN_Activities NOT IN ('', 'N/A', '0') THEN 'Primary Sector (Agri/Forest/Res)'
            ELSE 'Unclassified'
        END AS Detailed_Sector_Group,
        `Other Issues`,
        Access_premises,
        Employee_Size
    FROM business_data
) AS subquery
WHERE 
    -- Condition 1: Include businesses that haven't expanded
    (Employee_Size NOT IN ('6-10 employees', 'Above 10', '6-10', 'Above 10')
     OR id = 41) -- Exception for the specific scaled business with issues
    AND 
    (
      (`Other Issues` IS NOT NULL AND `Other Issues` NOT IN ('N/A', '', 'None', '0')) 
      OR 
      (Access_premises IS NOT NULL AND Access_premises NOT IN ('N/A', '', 'None', '0'))
    );

SELECT* 
FROM business_stagnation_voices;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
18 rows affected.


,id,Place,Detailed_Sector_Group,Other Issues,Access_premises
0,10,Lower Marindi,Retail & General Trade,2(depts),N/A
1,14,Wasweta 1,Retail & General Trade,N/A,2
2,17,Lower Marindi,Industrial/Manufacturing,Haggling customers,N/A
3,22,Lower Marindi,Retail & General Trade,Insecurity,N/A
4,23,Lower Marindi,Retail & General Trade,N/A,2
5,27,Wasweta 1,Retail & General Trade,N/A,3
6,29,Wasweta 1,Retail & General Trade,N/A,3
7,34,Wasweta 1,Informal/Small Scale,N/A,2
8,41,Lower Marindi,Hospitality & Leisure,Low flow of customers,3
9,46,Wasweta 1,Retail & General Trade,N/A,2


### Thematic Categorization(PROBLEM TREE)
Why certain businesses are stuck (stagnation) while others are struggling to maintain their scale.


In [28]:
%%sql
SELECT 
    Category,
    GROUP_CONCAT(id ORDER BY id SEPARATOR ', ') AS Specific_IDs,
    CASE 
        WHEN Category = 'Environmental Design' THEN 'Safety & Flow: "Insecurity" and "Low flow" suggest poor street lighting and a lack of pedestrian-friendly paths.'
        WHEN Category = 'Accessibility Crisis' THEN 'Premises Connectivity: Scores of 1 or 2 in Access_premises indicate physical difficulty or unappealing entry for customers.'
        WHEN Category = 'Market Volatility' THEN 'Economic Resilience: "Haggling" and "Volatility" suggest a lack of a stable, formalized market environment.'
        ELSE 'General Operational Constraints'
    END AS Planning_Issue_Identified
FROM (
    SELECT 
        id,
        CASE 
            -- Identifying Environmental Design (Insecurity or Low Flow)
            WHEN `Other Issues` LIKE '%Insecurity%' OR `Other Issues` LIKE '%Low flow%' THEN 'Environmental Design'
            
            -- Identifying Accessibility Crisis (Access_premises score of 1 or 2)
            WHEN Access_premises IN ('1', '2') THEN 'Accessibility Crisis'
            
            -- Identifying Market Volatility (Haggling, Volatility, or specific Depts issues)
            WHEN `Other Issues` LIKE '%Haggling%' OR `Other Issues` LIKE '%Volatility%' OR `Other Issues` LIKE '%depts%' THEN 'Market Volatility'
            
            ELSE 'Other Issues'
        END AS Category
    FROM business_data
    WHERE 
        -- Ensuring we only categorize rows with actual qualitative data
        ((`Other Issues` IS NOT NULL AND `Other Issues` NOT IN ('N/A', '', 'None', '0')) 
        OR 
        (Access_premises IS NOT NULL AND Access_premises NOT IN ('N/A', '', 'None', '0')))
) AS Categorized_Data
GROUP BY Category;

 * mysql+pymysql://root:***@localhost:3306/business_data
4 rows affected.


,Category,Specific_IDs,Planning_Issue_Identified
0,Accessibility Crisis,"14, 23, 34, 46, 49, 51, 52, 56, 60, 61, 62",Premises Connectivity: Scores of 1 or 2 in Acc...
1,Environmental Design,"22, 41","Safety & Flow: ""Insecurity"" and ""Low flow"" sug..."
2,Market Volatility,"10, 17, 47","Economic Resilience: ""Haggling"" and ""Volatilit..."
3,Other Issues,"18, 27, 29, 64",General Operational Constraints


### Planning Interventions Matrix

In [30]:
%%sql
CREATE OR REPLACE VIEW planning_intervention_matrix AS
SELECT 
    Category,
    Planning_Issue_Identified,
    CASE 
        WHEN Category = 'Environmental Design' THEN '1. Install high-mast street lighting in Lower Marindi to curb insecurity. 2. Redesign the A1 corridor to prioritize pedestrian walkways and safe "stop-over" points.'
        WHEN Category = 'Accessibility Crisis' THEN '1. Implement a "Frontage Improvement Program" to clear encroachments. 2. Pave drainage-sensitive access points in Wasweta 1 to ensure year-round business access.'
        WHEN Category = 'Market Volatility' THEN '1. Establish a Migori Business Resource Center to provide market data and price guidance. 2. Implement zoning for "Sector Hubs" to reduce retail oversaturation.'
        ELSE 'General Urban Management Improvements'
    END AS Proposed_Planning_Interventions
FROM (
    SELECT 
        CASE 
            WHEN `Other Issues` LIKE '%Insecurity%' OR `Other Issues` LIKE '%Low flow%' THEN 'Environmental Design'
            WHEN Access_premises IN ('1', '2') THEN 'Accessibility Crisis'
            WHEN `Other Issues` LIKE '%Haggling%' OR `Other Issues` LIKE '%Volatility%' OR `Other Issues` LIKE '%depts%' THEN 'Market Volatility'
            ELSE 'Other'
        END AS Category,
        CASE 
            WHEN `Other Issues` LIKE '%Insecurity%' OR `Other Issues` LIKE '%Low flow%' THEN 'Safety & Flow issues'
            WHEN Access_premises IN ('1', '2') THEN 'Physical Premises Connectivity'
            WHEN `Other Issues` LIKE '%Haggling%' OR `Other Issues` LIKE '%Volatility%' OR `Other Issues` LIKE '%depts%' THEN 'Economic Resilience'
            ELSE 'General Constraints'
        END AS Planning_Issue_Identified
    FROM business_data
    WHERE (`Other Issues` IS NOT NULL AND `Other Issues` NOT IN ('N/A', '', 'None')) 
       OR (Access_premises IS NOT NULL AND Access_premises NOT IN ('N/A', '', 'None'))
) AS Sub
GROUP BY Category, Planning_Issue_Identified;

SELECT * FROM planning_intervention_matrix;

 * mysql+pymysql://root:***@localhost:3306/business_data
0 rows affected.
4 rows affected.


,Category,Planning_Issue_Identified,Proposed_Planning_Interventions
0,Market Volatility,Economic Resilience,1. Establish a Migori Business Resource Center...
1,Accessibility Crisis,Physical Premises Connectivity,"1. Implement a ""Frontage Improvement Program"" ..."
2,Other,General Constraints,General Urban Management Improvements
3,Environmental Design,Safety & Flow issues,1. Install high-mast street lighting in Lower ...


## Executive Summary
### Spatial-Economic Dynamics of Migori Town
This study conducted a comprehensive spatial-economic analysis of Migori Town, utilizing data from over 60 business entities across three key character areas: Lower Marindi (CBD), Wasweta 1, and Oruba. By integrating quantitative SQL metrics with qualitative human voice feedback, the research identifies the critical friction points preventing Migori from transitioning from a retail-heavy informal economy to a high-growth specialized hub.

### Key Findings
1. The Growth Paradox: Scale vs. Stagnation
The analysis reveals a **growth ceiling** effect. While the Retail and General Trade sector dominates the landscape by volume (particularly in Wasweta 1), it suffers from the lowest expansion rates (approx.13%) due to extreme market saturation and hyper-competition. Conversely, specialized sectors like Logistics, Tech, and Manufacturing in Lower Marindi demonstrate a higher propensity for growth (up to 75% expansion) but are physically constrained by a lack of specialized infrastructure and high operating costs.

2. The Role of Governance & Gender
A significant finding in the demographic analysis showed that Male-owned businesses currently hold a higher share of "Specialized/High-Growth" sectors, while Female-owned businesses are concentrated in the "Retail/Survivalist" loop. Furthermore, the data suggests that Employees provide a more critical assessment of urban failure (citing insecurity and access) than Owners, who tend to focus on financial hurdles like taxes and rent.

3. The Accessibility & Security Crisis
The Human Voice(Other Issues) highlighted a major spatial failure:  

**Connectivity**:Over 70% of stagnating businesses rated their physical accessibility at a 2/5 or lower.  

**Environmental Safety**: In Lower Marindi, even businesses with Above 10 employees reported **Low Customer Flow** and **Insecurity** as primary threats to their sustainability. This indicates that poor urban design (lighting and pedestrian paths) is actively devaluing the CBD’s economic potential.  

### Planning Recommendations
To revitalize Migori Town’s economic landscape, the study proposes a Data-Driven Intervention Matrix:

1. Physical Infrastructure: Immediate paving and drainage improvement in Wasweta 1 to break the "Accessibility Crisis."

2. Public Safety: Implementation of high-mast lighting and pedestrian-centric "A1 Corridor" redesign in Lower Marindi to facilitate 24-hour economy flow.

3. Economic Policy: Transitioning from general licensing to Sector-Specific Zoning, encouraging the formation of "Specialized Hubs" to reduce retail oversaturation.

The study concludes that Migori Town’s stagnation is not merely a financial issue but a spatial one. The evidence shows that businesses are ready to scale, but the current urban fabric characterized by poor access, insecurity, and chaotic zoning acts as a physical barrier to expansion.